### PR TITLE
feat: Autogrowing Textarea 

### PR DESCRIPTION
**Pull request by istarkov**
## Description

Supports max-height and scrolling with our ScrollArea control see storybook

https://638245578b47d4399fba191a-pgsxdcimyt.chromatic.com/?path=/story/library-autogrow-text-area--demo

image

## Steps for reproduction

https://638245578b47d4399fba191a-pgsxdcimyt.chromatic.com/?path=/story/library-autogrow-text-area--demo

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `builder/env-check.js` if mandatory

### CODE REVIEW

File: packages/design-system/src/components/autogrow-text-area.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+            state,
+            className,
+          })}
+        >
+          {textValue}{" "}
+        </div>
+        <textarea
+          rows={1}
+          value={textValue}
+          onChange={(e) => setTextValue(e.target.value)}
```
Comment (kof) [inline packages/design-system/src/components/autogrow-text-area.tsx:84]:
```suggestion
          onChange={(event) => setTextValue(event.target.value)}
```

File: packages/design-system/src/components/autogrow-text-area.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  css?: CSS;
+  state?: "invalid";
+  value?: string;
+  defaultValue?: string;
+  onChange?: (value: string) => void;
+};
+
+export const AutogrowTextArea = forwardRef(
+  (
+    { css, className, state, value, defaultValue, onChange, ...props }: Props,
```
Comment (kof) [inline packages/design-system/src/components/autogrow-text-area.tsx:60]:
```suggestion
    { css, className, state, value: initialValue, defaultValue, onChange, ...props }: Props,
```

File: packages/design-system/src/components/autogrow-text-area.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    ref: Ref<HTMLTextAreaElement>
+  ) => {
+    // use
+    const [textValue, setTextValue] = useControllableState({
```
Comment (kof) [inline packages/design-system/src/components/autogrow-text-area.tsx:64]: once intiial value renamed these can be just value and setValue

**Review by kof — APPROVED** [review]

File: packages/design-system/src/components/autogrow-text-area.tsx (hunk 2 above)
Comment (istarkov) [reply packages/design-system/src/components/autogrow-text-area.tsx:60]: Its not initial value, it's value, and it's used as is if passed

**Pull request merged**